biz.aQute.bndlib/src/aQute/bnd/util/repository/DownloadListenerPromise.java

```diff
@@ -59,7 +58,7 @@ public DownloadListenerPromise(Reporter r, String task, Promise<File> promise,
 		})
 			.onFailure(failure -> {
 				logger.debug("{}: failure", this, failure);
-				String reason = Exceptions.toString(failure);
+				String reason = failure.toString();
```

Comment:
This seems to be eliding useful information when a failure occurs. Many times `Throwable.toString()` does not provide enough information to understand what happened which is why this code was changed to use `Exceptions.toString`.

Reply:
This prints the stack trace and in this place communication errors are by far the likeliest and then it creates zillions of stracktrace lines, not very useful. Better to see the actual error